python/pyspark/sql/tests/test_collection.py

```diff
@@ -289,6 +291,28 @@ def check_to_pandas_for_array_of_struct(self, is_arrow_enabled):
         else:
             self.assertEqual(type(pdf["array_struct_col"][0]), list)
 
+    def check_to_pandas_for_empty_df_with_nested_array_columns(self):
+        # SPARK-51112: Segfault must not occur when converting empty DataFrame with nested array
+        # columns to pandas DataFrame.
+        df = self.spark.createDataFrame(
+            data=[],
+            schema=StructType(
+                [
+                    StructField(
+                        name="b_int",
+                        dataType=IntegerType(),
+                        nullable=False,
+                    ),
+                    StructField(
+                        name="b",
+                        dataType=ArrayType(ArrayType(StringType(), True), True),
+                        nullable=True,
+                    ),
+                ]
+            ),
+        )
+        df.toPandas()
```

Comment:
Shall we check the generated pandas DataFrame is actually empty using `pyspark.testing.assertDataFrameEqual`?